revolv/static/js/base/accounting.js

```diff
@@ -0,0 +1,388 @@
+	// makes an AJAX call to the accountingJSON view when the selectors are changed
+	
+	$(".added").remove();
+	var changed_selector = $(this);
+	var changed_data = {};
+
+	if (changed_selector.hasClass("start_selector")) {
+		changed_data['start_date'] = changed_selector[0].value;
+	} else if (changed_selector.hasClass("end_selector")) {
+		changed_data['end_date'] = changed_selector[0].value;
+	} else {
+		changed_data['project_choice'] = changed_selector[0].value;
+	}
+	console.log(changed_data)
+	makeAJAXcall(changed_data);
+});
+
+/**
+ * Makes the AJAX Call
+ * @param {Object} changed_data - An object with either 0 or 1 keys that maps a selector to its new value.
+ */
+function makeAJAXcall(changed_data) {
+	var csrftoken = getCookie('csrftoken');
+
+	$.ajax({
+	    url: window.JSONURL,
+	    method: "GET",
+	    data: changed_data,
+	    headers: {"X-CSRFToken": csrftoken},
+	    success:function(data, textStatus, xhr){
+	    	setValues(data);
+	    },
+	    error:function (xhr, textStatus, thrownError){
+	        alert("Sorry, something went wrong! Try again?");
+	    },
+	});
+}
+
+/**
+ * Sets the time values, creates column headers, creates row headers, and fills in table values.
+ * @param {Object} data - The JSON returned by the AJAX call.
+ */
+function setValues(data) {
+	setTimeValues(data);
+	makeColumnHeaders(data);
+	makeRowHeaders(data);
+	setTableValues(data);
+	setTotalsAndCashFlow(data);
+}
+
+/**
+ * Fills in the appropriate table cells from the given data.
+ * @param {Object} data - The JSON returned by the AJAX call.
+ */
+function setTableValues(data) {
+	placeElements(data, 'cash_in', 'donations', placeDonationElement);
+	placeElements(data, 'cash_in', 'repayments', placeROIElement);
+	placeElements(data, 'cash_in', 'adjustments', placeCashInAdjustmentElement);
+	placeElements(data, 'cash_out', 'reinvestments', placeAdminInvestmentElement);
+	placeElements(data, 'cash_out', 'adjustments', placeCashOutAdjustmentElement);
+}
+
+/**
+ * Cleanes up the String to be JQuery friendly.
+ * @param {String} text - The text to be cleaned up to be made JQuery friendly.
+ */
+function cleanUpString(text) {
+	text = text.split(' ').join('-').split(';').join('sm').split(":").join('co').split("'").join("ap").split("!").join('ex');
+	return text;
+}
+
+/**
+ * Places a donation element on the page.
+ * @param {Object} data_to_insert - A dictionary with values to insert,
+ * @param {String} project_name - The name of the project we are inserting under.
+ * @param {String} time_key - The time period in which this was inserted.
+ */
+function placeDonationElement(data_to_insert, project_name, time_key) {
+	var total = data_to_insert['total'];
+	var payment_service_fees = data_to_insert['payment_service_fees'];
+	var retained_donations = data_to_insert['retained_donations'];
+	var className  = '.project-' + cleanUpString(project_name);
+	$('.total-donation').filter(className).append('<td class="added">' + String(total) + "</td>");
```

Comment:
Casting to string like this is unneeded in javascript, since you're adding it to a string is will get automatically converted. Also isn't it a string already anyway?